src/components/button/button_icon/button_icon.tsx

```diff
@@ -173,6 +180,15 @@ export const EuiButtonIcon: FunctionComponent<Props> = ({
     );
   }
 
+  // `original` size doesn't exist in `EuiLoadingSpinner`
+  // when the `iconSize` is `original` we don't pass any size to the `EuiLoadingSpinner`
+  // so it gets the default size
+  const loadingSize = iconSize === 'original' ? undefined : iconSize;
+
+  if (iconType && isLoading) {
+    buttonIcon = <EuiLoadingSpinner size={loadingSize!} />;
```

Comment:
`undefined` should be a valid input for size - you should be able to remove the `!` without issue
```suggestion
    buttonIcon = <EuiLoadingSpinner size={loadingSize} />;
```